CRM/Contribute/Form/ContributionBase.php

```diff
@@ -515,22 +515,12 @@ private function initSet($form) {
     $priceSetId = CRM_Price_BAO_PriceSet::getFor('civicrm_contribution_page', $this->_id);
     //check if price set is is_config
     if (is_numeric($priceSetId)) {
-      if (CRM_Core_DAO::getFieldValue('CRM_Price_DAO_PriceSet', $priceSetId, 'is_quick_config') && $form->getVar('_name') != 'Participant') {
-        $form->assign('quickConfig', 1);
-      }
+      $form->assign('quickConfig', CRM_Core_DAO::getFieldValue('CRM_Price_DAO_PriceSet', $priceSetId, 'is_quick_config'));
     }
     // get price info
     if ($priceSetId) {
-      if ($form->_action & CRM_Core_Action::UPDATE) {
```

Comment:
@eileenmcnaughton I'm just thinking isn't this needed when your using a contribution page to pay an invoice?

Reply:
@seamuslee001 yeah - but check my update above - if that IF is entered it passes the ContributionPage Id to the next function - which expects contribution id or participant id - so it never makes sense

(I think we do something else when paying an invoice)

Reply:
@seamuslee001 just trying to get this one finished off (I don't think I'll do the follow patches now as I don't think I remember what they were any more - but at least this part won't need re-visiting...)

This line

``` $form->_values['line_items'] = CRM_Price_BAO_LineItem::getLineItems($form->_id, 'contribution');```

only makes sense when `id` is a contribution. However, id is a ContributionPage ID. I also checked nothing crazy was happening in the 'Pay Now' context from the user dashboad  - but in that case the mysterious `ccid` is the `contribution_id` and `id` is not messed with.

![image](https://user-images.githubusercontent.com/336308/216875987-75f4dd05-ddf8-4a6f-80ba-b9402e30ab2d.png)

I did try an r-run on this and it worked fine once I realised that without the contribution page setting being set it advertises a broken url - (That is a separate issue & I am gonna leave it alone since I'm currently trying to reduce the number of open PRs I have)